com.unity.render-pipelines.high-definition/Runtime/Debug/DebugDisplay.hlsl

```diff
@@ -83,6 +83,14 @@ void GetPropertiesDataDebug(uint paramId, inout float3 result, inout bool needLi
             result = float3(1.0, 0.0, 0.0);
 #else
             result = float3(0.0, 0.0, 0.0);
+#endif
+            break;
+
+        case DEBUGVIEWPROPERTIES_DEFERRED_MATERIALS:
+#ifdef _DEFERRED_CAPABLE_MATERIAL
+            result = float3(0.0, 1.0, 0.0);
```

Comment:
So you miss to take into account the FrameSettings.
In FrameSettings we can disable or not the render in deferred. If FrameSettings are setup to forward then we should show material as forward.
result = debugDisplay.IsLitShaderModeDeferred ? float3(0.0, 1.0, 0.0) : float3(1.0, 0.0, 0.0);

And you need to add it to the command buffer.

+ TerrainLit as you mention :)

Reply:
I intentionally didn't consider the FrameSettings for the visualization. I.e. if the purpose of this mode is to inform user which materials _can_ be rendered in deferred, but not necessarily which ones are rendered in deferred for the current camera. So that user can see how optimized the scene is for deferred rendering regardless of the active mode. I can change this and update the comment though, if it's preferred to obey the active mode instead.

Reply:
FYI, after discussion with artists changed the visualization to take FrameSettings into account